src/main/java/frc/robot/subsystems/Maflipulator.java

```diff
@@ -0,0 +1,193 @@
+package frc.robot.subsystems;
+
+import com.chopshop166.chopshoplib.outputs.SendableSpeedController;
+
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.GenericHID.Hand;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.InstantCommand;
+import edu.wpi.first.wpilibj.command.PIDCommand;
+import edu.wpi.first.wpilibj.command.Scheduler;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.interfaces.Potentiometer;
+import frc.robot.Robot;
+import frc.robot.RobotMap;
+
+public class Maflipulator extends Subsystem {
+
+    public enum MaflipulatorSide {
+        kFront, kBack;
+    }
+
+    private final static double FRONT_LOWER_ANGLE = 70;
+    private final static double FRONT_UPPER_ANGLE = 180;
+    private final static double BACK_LOWER_ANGLE = 290;
+    private final static double BACK_UPPER_ANGLE = 180;
+
+    private final static double FLIP_MOTOR_SPEED = 1;
+
+    private MaflipulatorSide currentPosition;
+
+    private SendableSpeedController flipMotor;
+    private Potentiometer anglePot;
+
+    double angleCorrection;
+    PIDController anglePID;
+
+    public Maflipulator(final RobotMap.MaflipulatorMap map) { // NOPMD
+        super();
+        flipMotor = map.getFlipMotor();
+        anglePot = map.getMaflipulatorPot();
+
+        anglePID = new PIDController(.01, .0009, 0.0, 0.0, anglePot, (double value) -> {
+            angleCorrection = value;
+        });
+
+        if (anglePot.get() < 180)
+            currentPosition = MaflipulatorSide.kFront;
+        else
+            currentPosition = MaflipulatorSide.kBack;
+
+    }
+
+    @Override
+    public void initDefaultCommand() {
+        // Set the default command for a subsystem here.
+        setDefaultCommand(restrictRotate());
+    }
+
+    public Command manualFlip() {
```

Comment:
`manualFlip` doesn't seem to be used anymore, or serve a real purpose in favor of `restrictRotate`. Recommend deleting